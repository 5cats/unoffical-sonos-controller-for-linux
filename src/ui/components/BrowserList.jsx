
import port from '../port';
import model from '../model';

import BrowserListItem from './BrowserListItem';

import React from 'react/addons';
import { Cursor, ImmutableOptimizations }  from 'react-cursor';
import EventableMixin from '../mixins/EventableMixin';

var initialState = {
	source: null,
	searchType: null,
	headline: 'Select a Music Source',
	items: [
		{
			title: 'Sonos Favourites',
			source: 'favourites'
		},
		{
			title: 'Music Library',
			source: 'library'
		}
	]
};

var librarySearch= {
	headline: 'Browse Music Library',
	source: 'library',
	items: [
		{
			title: 'Artists',
			searchType: 'artists'
		},
		{
			title: 'Albums',
			searchType: 'albums'
		},
		{
			title: 'Composers',
			searchType: 'composers'
		},
		{
			title: 'Genres',
			searchType: 'genres'
		},
		{
			title: 'Tracks',
			searchType: 'tracks'
		},
		{
			title: 'Playlists',
			searchType: 'playlists'
		}
	]
};

var rewinds = [];


class BrowserList {

	getInitialState () {
		rewinds.push(initialState);
		return initialState;	
	}

	render () {

		var self = this;
		var clickAction

		var listItemNodes = this.state.items.map(function (i, p) {
			var position = p + 1;
			return (
				<BrowserListItem data={i} position={position} clickAction={self._clickAction} />
			);
		});

		return (
			<div id="music-sources-container">
				<h4>{this.state.headline}</h4>
				<ul id="browser-container">
					{{listItemNodes}}
				</ul>
			</div>
		);
	}

	_clickAction (e, item) {
		console.log(item);

		if(!this.state.source) {
			if(item.source === 'library') {
				this.setState(librarySearch);
				rewinds.push(this.state);
			} else {
				this.setState(initialState);
			}			
		}

		if(this.state.source === 'library' && item.searchType) {
			console.log('here', item);

			port.postMessage({
				type: 'browse',
				host: model.coordinator.host,
				searchType: item.searchType,
			});
		}
	}
}

BrowserList.prototype.displayName = "BrowserList";
BrowserList.prototype.mixins = [
	ImmutableOptimizations(['cursor']),
	EventableMixin
];
export default React.createClass(BrowserList.prototype);