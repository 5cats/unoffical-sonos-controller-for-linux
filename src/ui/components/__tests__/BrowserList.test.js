import React from 'react';
import { BrowserList } from '../BrowserList';
import { render } from 'enzyme';

jest.mock('../BrowserListItem', () => () => <p />);

describe('BrowserList', () => {
    it('renders and matches snapshot', () => {
        const props = {
            currentState: {
                items: [
                    {
                        id: 1
                    }
                ],
                title: 'My List',
                sournce: 'source'
            },
            history: []
        };

        const context = render(<BrowserList {...props} />);
        expect(context).toMatchSnapshot();
    });
});
