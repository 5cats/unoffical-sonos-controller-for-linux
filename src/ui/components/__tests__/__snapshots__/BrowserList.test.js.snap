// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrowserList renders and matches snapshot 1`] = `
<div
  id="music-sources-container"
>
  <h4>
    My List
  </h4>
  <ul
    id="browser-container"
  >
    <div
      style="overflow:visible;height:0;width:0"
    />
  </ul>
</div>
`;
